draft-ietf-tls-esni.md

```diff
@@ -1012,39 +985,60 @@ server to include any information it requires to process the second ClientHello.
 
 ## Backend Server {#backend-server}
 
-Upon receipt of an "ech_is_inner" extension in a ClientHello, if the backend
-server negotiates TLS 1.3 or higher, then it MUST confirm ECH acceptance to the
-client by computing its ServerHello as described here.
+Upon receipt of an "encrypted_client_hello" extension of type `inner` in a
+ClientHello, if the backend server negotiates TLS 1.3 or higher, then it MUST
+confirm ECH acceptance to the client by computing its ServerHello as described
+here.
 
-The backend server begins by generating a message ServerHelloECHConf, which is
-identical in content to a ServerHello message with the exception that
-ServerHelloECHConf.random is equal to 24 random bytes followed by 8 zero bytes.
-It then computes an 8-byte string
+The backend server embeds in ServerHello.random a string derived from the inner
+handshake. It begins by computing its ServerHello as usual, except the last 8
+bytes of ServerHello.random are set to zero. It then computes the transcript
+hash for ClientHelloInner up to and including the modified ServerHello, as
+described in {{RFC8446}}, Section 4.4.1. Let transcript_ech_conf denote the
+output. Finally, the backend server overwrites the last 8 bytes of the
+ServerHello.random with the following string:
 
 ~~~
-   accept_confirmation = HKDF-Expand-Label(0,
+   accept_confirmation = HKDF-Expand-Label(
+      HKDF-Extract(0, ClientHelloInner.random),
```

Comment:
Why is `HKDF-Extract(0, ClientHelloInner.random)` now necessary?

Reply:
It shouldn't be -- we can probably drop it.

Reply:
Using `0` as the PRK isn't safe, since this value is supposed to be pseudorandom (see https://datatracker.ietf.org/doc/html/rfc5869). Assuming `ClientHelloInner.random` has entropy,  `HKDF-Extract(0, ClientHelloInner.random)` is pseudorandom. We might be able to get away with using `ClientHelloInner.random` as the PRK directly, if the client generates this value securely. Note that this is change matches what we had in draft-ietf-tls-esni-08.

Reply:
HKDF is a dual-PRF, so the result is the same. We don't need to do this.

Reply:
I don't think that's true. In any case, the change is orthogonal, so I'll move it to a separate PR.